Xamarin.Forms.Xaml.UnitTests/DataTemplateExtension.xaml.cs

```diff
@@ -0,0 +1,33 @@
+﻿using NUnit.Framework;
+
+using Xamarin.Forms.Core.UnitTests;
+
+namespace Xamarin.Forms.Xaml.UnitTests
+{
+	public partial class DataTemplateExtension : ContentPage
+	{
+		public DataTemplateExtension() => InitializeComponent();
+		public DataTemplateExtension(bool useCompiledXaml)
+		{
+			//this stub will be replaced at compile time
+		}
+
+		[TestFixture]
+		class Tests
```

Comment:
Is this something we like? Nested test classes?

Reply:
1/ it's something I like
2/ it's the most convenient way in this type of cases, so we can access `layout.label` which is a private field created by x:Name
3/ all xaml unit tests uses the same schema
4/ and now, for longer answer. in the distant pasts I can recall working professionally with unit tests (20 years ago, java environment), unit tests were always placed in nested classes of the class being tested. That allowed to make real unit tests for the private part of the class. Years passed, and we tend to move unit tests in their own assemblies. And those new kind of unit tests no longer test the way a class works, but they test more the public API of it. I'm not nostalgic here, but we lost some finer grained testing in the transition. And we're using `internal` and `InternalsVisibleTo` when we want that agility back...